account_avatax_exemption/models/exemption.py

```diff
@@ -0,0 +1,317 @@
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError, ValidationError
+
+
+class ExemptionRule(models.Model):
+    _name = "exemption.code.rule"
+    _description = "Avatax Custom Rules"
+
+    name = fields.Char(index=True, default=lambda self: _("New"))
+    state = fields.Selection(
+        [
+            ("draft", "Draft"),
+            ("progress", "In Progress"),
+            ("done", "Done"),
+            ("cancel", "Cancelled"),
+        ],
+        default="draft",
+    )
+    exemption_code_id = fields.Many2one(
+        "exemption.code", string="Entity Use Code", required=True
+    )
+    state_id = fields.Many2one(
+        "res.country.state",
+        string="Region",
+    )
+    avatax_id = fields.Char("Avatax Rule ID", readonly=True, copy=False)
+    avatax_tax_code = fields.Many2one("product.tax.code")
+    is_all_juris = fields.Boolean(default=True)
+    avatax_rate = fields.Float()
+    taxable = fields.Boolean()
+
+    @api.constrains("avatax_rate")
+    def _check_avatax_rate(self):
+        """
+        Prevent the Avatax rate with range 0 to 100
+        """
+        for record in self:
+            if record.avatax_rate < 0 or record.avatax_rate > 100:
+                raise ValidationError(_("Avatax rate range is from 0 to 100"))
```

Comment:
```suggestion
                raise ValidationError(self.evn._("Avatax rate range is from 0 to 100"))
```

> You can replace the call to translate a string (`_`) with `self.env._` for getting some performance improvement in some cases. See https://github.com/odoo/odoo/pull/174844.

Similar to other cases in this module